### PR TITLE
snap/pack, cmd/snap: add `snap pack --check-skeleton`

### PR DESCRIPTION
**Pull request by bboozzoo**
Add support for `snap pack --check-skeleton`. Extend `snap pack` test suite.

### CODE REVIEW

File: cmd/snap/cmd_pack.go [hunk 1]
```diff
@@ -56,6 +60,14 @@ func (x *packCmd) Execute([]string) error {
 		x.Positional.TargetDir = "."
 	}
 
+	if x.CheckSkeleton {
+		_, err := pack.LoadValidate(x.Positional.SnapDir)
+		if err == snap.ErrMissingPaths {
```
Comment (mvo5) [inline cmd/snap/cmd_pack.go]: Heh, this is surprisingly straightforward!

File: cmd/snap/cmd_pack.go [hunk 2]
```diff
@@ -45,7 +47,9 @@ func init() {
 		longPackHelp,
 		func() flags.Commander {
 			return &packCmd{}
-		}, nil, nil)
+		}, map[string]string{
+			"check-skeleton": i18n.G("Validate snap directory only"),
```
Comment (chipaca) [inline cmd/snap/cmd_pack.go]: i feel that description does a poor job of explaining the flag.
At the same time, explaining the flag is probably more text than should be in a flag's  help.
So: please explain what the flag does in detail, in the command's help; then a short flag description that relies on that explanation will be appropriate (but even then I don't think this is the right text).

File: snap/pack/pack.go [hunk 3]
```diff
@@ -211,24 +211,31 @@ func copyToBuildDir(sourceDir, buildDir string) error {
 	})
 }
 
-func prepare(sourceDir, targetDir, buildDir string) (snapName string, err error) {
+// LoadValidate attempts to load and validate  snap data from source directory
+func LoadValidate(sourceDir string) (*snap.Info, error) {
```
Comment (chipaca) [inline snap/pack/pack.go]: I'm very bad with names, but this is a bad name :-)

**Comment by chipaca** [comment]
FWIW [this](https://www.bricklink.com/v2/catalog/catalogitem.page?S=4841) is the _real_ #4841.

File: snap/pack/pack.go (hunk 3 above)
Comment (bboozzoo) [reply snap/pack/pack.go]: Hmm I could rename it to `func CheckSkeleton(sourceDir string) error` and bury the ugly name in the package :)

File: cmd/snap/cmd_pack.go [hunk 4]
```diff
@@ -25,27 +25,41 @@ import (
...
+The pack command packs the given snap-dir as a snap and writes the result to
+target-dir. If target-dir is omitted, the result is written to current
+directory. If both source-dir and target-dir are omitted, the pack command packs
+the current directory.
+
+When used with --check-skeleton, pack only checks whether snap-dir contains
+valid snap metadata and raises an error otherwise. Application commands listed
+in snap metadata file, but appearing with incorrect permission bits or missing
+from snap-dir are signaled through diagnostic messages and do not result in an
+error.
```
Comment (chipaca) [inline cmd/snap/cmd_pack.go]: Application commands listed in the snap metadata but _missing_ won't result in an error, but incorrect permission bits will.

Comment (bboozzoo) [reply cmd/snap/cmd_pack.go]: Silly me, thanks for catching this.

**Pull request merged**